src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1285,6 +1285,85 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
             }
         }
 
+        internal bool HasCollectionExpressionAddMethod(SyntaxNode syntax, TypeSymbol targetType)
+        {
+            const string methodName = "Add";
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+
+            var lookupResult = LookupResult.GetInstance();
+            LookupInstanceMember(lookupResult, targetType, leftIsBaseReference: false, rightName: methodName, rightArity: 0, invoked: true, ref useSiteInfo);
+            bool anyApplicable = lookupResult.IsMultiViable &&
+                lookupResult.Symbols.Any(s => s is MethodSymbol m && isApplicableAddMethod(m, expectingExtensionMethod: false));
+            lookupResult.Free();
+            if (anyApplicable)
+            {
+                return true;
+            }
+
+            var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
+            foreach (var scope in new ExtensionMethodScopes(this))
+            {
+                var methodGroup = MethodGroup.GetInstance();
+                PopulateExtensionMethodsFromSingleBinder(scope, methodGroup, syntax, implicitReceiver, rightName: methodName, typeArgumentsWithAnnotations: default, BindingDiagnosticBag.Discarded);
+                anyApplicable = methodGroup.Methods.Any(m => isApplicableAddMethod(m, expectingExtensionMethod: true));
+                methodGroup.Free();
+                if (anyApplicable)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+
+            static bool isApplicableAddMethod(MethodSymbol method, bool expectingExtensionMethod)
```

Comment:
>isApplicableAddMethod [](http://example.com/codeflow?start=24&length=21)

We probably should pay attention to use-site diagnostics, and make consumers of ```HasCollectionExpressionAddMethod``` aware of them in case of a failure.